asciidoc-confluence-publisher-converter/src/main/java/org/sahli/asciidoc/confluence/publisher/converter/AsciidocConfluencePage.java

```diff
@@ -242,14 +242,15 @@ private static Options options(Path templatesFolder, Path baseFolder, Path gener
                 .build();
     }
 
-    private static Function<String, String> replaceCrossReferenceTargets(Path pagePath, Map<String, Object> userAttributes, PageTitlePostProcessor pageTitlePostProcessor, Charset sourceEncoding, String spaceKey) {
+    private static Function<String, String> replaceCrossReferenceTargets(Path pagePath, Map<String, Object> userAttributes, PageTitlePostProcessor pageTitlePostProcessor, Charset sourceEncoding, String spaceKey, Options options) {
         return (content) -> replaceAll(content, PAGE_TITLE_PATTERN, (matchResult) -> {
             String htmlTarget = matchResult.group(1);
             Path referencedPagePath = pagePath.getParent().resolve(Paths.get(htmlTarget.substring(0, htmlTarget.lastIndexOf('.')) + ".adoc"));
 
             try {
+                options.setBaseDir(referencedPagePath.getParent().toString());
```

Comment:
It is just to be sure that the base dir is set to the one of included file is in. Just in case you are including a file in a diferent directory.

Reply:
Modifying `options` here (with potential side effects in case `options` would later be used outside of this method) does not seem very clean to me. Better would in my opinion be to re-construct (maybe based on a clone of `options`) the required state for processing the referenced page. What do you think?

Reply:
Yes, that's prob. much cleaner. I'll look into it!

Reply:
Sadly, the `Options` class dose not provide any getters intendet for external use. Therefore I am passing down all the properties used for creating a new instance. This results in some quite long method headers. What do you think?

Reply:
Sorry for the late reply @Marfien - that is indeed a pity! Cloning and modifying the existing options would have been very convenient. 

Anyway, going one step back: in which use case would setting the base dir here be relevant? Can we cover this with a specific test? If I just remove setting the base dir to the referenced page path, no test fails.

Reply:
I thought about the page might need to resolve some more included files relative to the page. The base dir is therefore the orgin for those relative paths in my understanding. Did I got that right?

Reply:
Yes, I guess you are right. It would be great to cover this case with an appropriate test case.

Regarding `Options.map()` and `Options(Map)`: both APIs are already deprecated and the constructor accepting a map has already been removed in the most current version of asciidoctorj. I was also thinking about using reflection for reading and setting the map, but that is not really future-proof ...

Reply:
What do you want to test in this case? If it fails if the basedir is wrong?

Regarding the `Options` map: I think the deprecation annotation is just beeing used to mark the method as internal and not ment to be used by third-party. But I might open a pull request on [asciidoctor/asciidoctorj](https://github.com/asciidoctor/asciidoctorj) for Options to implement `Cloneable`.

Reply:
Opening a pull request would of course be the cleanest option, thank you for considering this! Then we could optimize the code in Confluence Publisher after upgrading to the new asciidoctorj version.

Regarding the test: I would be very happy to have a test that fails if one would comment out the new line of code that sets the base dir to the referenced page. That should be easily possible with an appropriate test data setup (i.e. having the relevant page including another one). What do you think?

Reply:
I just created the issue in [asciidoctor/asciidoctorj#1270](https://github.com/asciidoctor/asciidoctorj/issues/1279).

Regarding the test: That is indeed a good idea. I'll take a look into it!

Reply:
Thank you for opening the issue for AsciidoctorJ. I commented there, too. For now, I would go with your current implementation, I think the test coverage is on our side :-) Ok?

Reply:
Yes, that's fine on my side. I'd add a comment linking to the issue at AsciidoctorJ to keep a note that it might can be change in the future. Or do you think that there should no open TODOs in the code?

Reply:
I think we are good the way it is now. Thank you!